src/System.IO/tests/Stream/Stream.ReadOnly.cs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    // This class emulates the behavior of Stream.Null,
+    // except CanWrite == false and attempting to write
+    // will throw a InvalidOperationException.
+    public sealed class ReadOnlyStream : Stream
```

Comment:
@stephentoub Good point; I'll switch to using the `MemoryStream` constructor that makes it a read-only stream.